build/flaws.js

```diff
@@ -479,7 +479,12 @@ function injectHeadingLinksFlaws(level, doc, $, rawContent) {
 async function fixFixableFlaws(doc, options, document) {
   if (!options.fixFlaws || document.isArchive) return;
 
-  let newRawHTML = document.rawHTML;
+  const { rawBody, isMarkdown } = document;
```

Comment:
Drive-by thought: Might this be a case for a Strategie Pattern?

Like, you have a BaseClass for handling flaw detection with some methods the strategy must implement.
Then, you derive a class for HTML flaw detection and another one for Markdown flaw detection.
The most method will remain in HTML flaw detection subclass (and the Markdown variant throwing errors), but you can move shared logic into the BaseClass if needed.

This way, the decision on „isMarkdown” happens on the caller side (to decide, which strategy class to use).